src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -1102,6 +1124,12 @@ window.qBittorrent.DynamicTable ??= (() => {
                 td.title = status;
             };
 
+            this.columns["status"].compareRows = function(row1, row2) {
+                const state1 = this.getRowValue(row1);
+                const state2 = this.getRowValue(row2);
+                return compareNumbers(this.sortOrder[state1], this.sortOrder[state2]);
```

Comment:
Is it possible to store the `status` order number in the row property?
This way we can avoid mapping from string to number.

Reply:
>but it would get mixed with other torrent information.

Would this work?
```js
                            if (response["torrents"][key]["state"]) {
                                const state = response["torrents"][key]["state"];
                                response["torrents"][key]["status"] = state;
                                response["torrents"][key]["#statusOrder"] = sortOrder[state];  // prefix `#` to indicate it is private
                            }
```
As long as `#statusOrder` won't be shown on screen it should be fine.

Reply:
We won't be able to access #statusOrder in compareRows function ("Uncaught SyntaxError: Private field '#statusOrder' must be declared in an enclosing class"). I guess we could just prefix it with underscore - would that be ok?

Reply:
>I guess we could just prefix it with underscore - would that be ok?

👍